huaweicloud/services/vod/resource_huaweicloud_vod_transcoding_template_group.go

```diff
@@ -0,0 +1,592 @@
+package vod
+
+import (
+	"context"
+	"log"
+	"regexp"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	vod "github.com/huaweicloud/huaweicloud-sdk-go-v3/services/vod/v1/model"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceTranscodingTemplateGroup() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceTranscodingTemplateGroupCreate,
+		ReadContext:   resourceTranscodingTemplateGroupRead,
+		UpdateContext: resourceTranscodingTemplateGroupUpdate,
+		DeleteContext: resourceTranscodingTemplateGroupDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		//request and response parameters
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 128),
+					validation.StringMatch(regexp.MustCompile(`^\w+$`),
+						"The name can only consist of letters, digits and underscores (_)."),
+				),
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"auto_encrypt": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"is_default": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"watermark_template_ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"common": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
```

Comment:
if the max-items = 1, can we flatten them without `common`?

Reply:
will fix this with other similar problem in another pr

Reply:
will fix this with other similar problem in another pr